src/Types.jl

```diff
@@ -219,6 +219,7 @@ Base.@kwdef mutable struct Compat
     val::VersionSpec
     str::String
 end
+Base.:(==)(t1::Compat, t2::Compat) = all(x -> (getfield(t1, x) == getfield(t2, x))::Bool, fieldnames(Compat))
```

Comment:
Do you need to define `hash` now as well? Also, seems easier to just write out the actual fields here.

Reply:
Should probably just compare the `val`?

Reply:
Sounds good. Both changes made